src/nvim/option.c

```diff
@@ -3029,10 +3013,6 @@ char *set_option_value(const char *const name, const long number, const char *co
                        const int opt_flags)
   FUNC_ATTR_NONNULL_ARG(1)
 {
-  if (is_tty_option(name)) {
-    return NULL;  // Fail silently; many old vimrcs set t_xx options.
```

Comment:
This will error now, right? That seems ok. Maybe we need a single-line test somehwere? `test/functional/options/defaults_spec.lua` or `test/functional/legacy/options_spec.lua`